core/core.go

```diff
@@ -40,6 +41,21 @@ func (c *Core) InsertChain(blocks types.Blocks) (int, error) {
 	domWait := false
 	for i, block := range blocks {
 		isCoincident := c.sl.engine.HasCoincidentDifficulty(block.Header())
+		// If we just mined this block, its possible the subordinate manifest in our
+		// block body is incorrect. If so, ask our sub for the correct manifest,
+		// update and rewrite the correct body.
+		if block.ManifestHash() != block.SubManifest().Hash() {
```

Comment:
How are we validating manifests for application? Here it just feels like a simple check of what's in the body compared to the header. Are we doing an ask every time to validate the subordinate manifest if we have that subClient?

Reply:
When our dom node starts the append process, he passes in the manifest hash stored in his header. We check here that the manifest hash the dom gave us matches the hash of the manifest we compute locally for this block:
https://github.com/dominant-strategies/go-quai/blob/5ec5fe9f33328f38b04c285c27c24c65bb02a90c/core/headerchain.go#L140